src/term/check/type_check.rs

```diff
@@ -0,0 +1,69 @@
+use core::fmt;
+use std::collections::HashMap;
+
+use crate::term::{Book, DefId, Name, RulePat};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Type {
+  Any,
+  Adt(Name),
+}
+
+type TypedDefinitions = HashMap<DefId, Vec<Type>>;
```

Comment:
Minor semantic thing, but this map doesn't hold the definitions, only their types, so a name like `DefinitionTypes` would be more appropriate